svgdigitizer/svgplot.py

```diff
@@ -143,67 +184,92 @@ def get_trafo(self, axis):
-                for key, val in label_patterns.items():
-                    regex_match = re.match(val, text.firstChild.firstChild.data)
-                    if isinstance(regex_match, re.Match):
-                        match = True
-
-                        try:
-                            labeled_paths[key].append((text, paths, regex_match))
-                        except KeyError:
-                            labeled_paths[key] = [(text, paths, regex_match)]
-
-                if not match:
-                    print(f"Label not obeying known patterns: \"{text.firstChild.firstChild.data}\"")
-
-                    # warn for ambiguous paths
-                    # gives false positive for scale bars
-                    #if len(paths) > 1:
-                    #    print("""Multiple paths grouped with single text!
-                    #          Only first will be taken into account. Please review supplied svg file!""")
-
-                #labeled_paths[text] = paths
-        # if more paths (lists) than texts exist it is likely that something is wrong
-        if len(labeled_paths) > len([text for text in texts]):
-            print("""Unlabeled paths found! Please review supplied svg file!""")
+        We only consider paths which are grouped with a label, i.e., a `<text>`
+        element. These text elements then tell us about the meaning of that
+        path, e.g., the path that is labeled as `curve` is the actual graph we
+        are going to extract the (x, y) coordinate pairs from.
+        """
+        labeled_paths = {key: [] for key in label_patterns}
+
+        groups = set(path.parentNode for path in self.doc.getElementsByTagName('path'))
+
+        for group in groups:
+            if group.nodeType != Node.ELEMENT_NODE or group.tagName != 'g':
+                logger.warning("Parent of <path> is not a <g>. Ignoring this path and its siblings.")
+                continue
+
+            # Determine all the <path>s in this <g>.
+            paths = [path for path in group.childNodes if path.nodeType == Node.ELEMENT_NODE and path.tagName == 'path']
+            assert paths
+
+            # Determine the label associated to these <path>s
+            label = None
+            for child in group.childNodes:
+                if child.nodeType == Node.COMMENT_NODE:
+                    continue
+                elif child.nodeType == Node.TEXT_NODE:
+                    if SVGPlot._text_value(child):
+                        logger.warning(f'Ignoring unexpected text node "{SVGPlot._text_value(child)}" grouped with <path>.')
+                elif child.nodeType == Node.ELEMENT_NODE:
+                    if child.tagName == 'path':
+                        continue
+                    if child.tagName != "text":
+                        logger.warning(f"Unexpected <{child.tagName}> grouped with <path>. Ignoring unexpected <{child.tagName}>.")
+                        continue
+
+                    if label is not None:
```

Comment:
`label` is always `None`

Reply:
No, it's set below where it says `label = child`. So this will complain when there are multiple `<text>` nodes in a group.